### PR TITLE
Fixes #37 Unathi horns and frills displayed as ERROR on preview as well as Tajaran and Skrell specific hair styles also showing the same error. Fixes a Unathi, Tajaran and Vulpkanin tail colouration issue. Fixes #26 wings and tails from the extras menu not showing when facing north.

### LINKED ISSUE #26: Species mods like tails and wings draw on the wrong layer when facing north.
https://i.imgur.com/52eIeAd.png Wings are drawing on the wrong layer I think, at least for when you are facing north.

### LINKED ISSUE #37: Unathi horns/hair options slap a big fat Error onto the preview sprite.
When customizing a unathi character, choosing any of the Unathi horn types slaps a error tag across the preview sprite
Unathihairerror1

This goes for every single hairstyle option for unathi horns.

Changing the colour of the hair/horn style does change the colour of the error text.

### PR DESCRIPTION
**Pull request by ShadzFen**
## About The Pull Request

This branch fixes issue #37 detailing sprite problems regarding the Unathi horns, frills and other species-exclusive hairstyles. It also adds a new .dmi file organizing all Unathi extras, as well as some additional tails, tail spikes, and body textures that could possibly be used in the future as extras if an extras menu was ever added.

This branch now also fixes base Unathi, Tajaran and Vulpkanin tail colouration issues. Tails now show the correct colour selected with no issues.

This branch now also fixes #26 detailing sprite problems regarding tails and wings from the Species Customization menu not displaying on a higher layer when facing north.

## Why It's Good For The Game

It fixes 3 ongoing issue and adds potential to further increase character customization in the future, while remaining easy to find all the extras in a single file.

## Changelog

**Added** 
- A few new horns and frills styles for Unathi. 
- Added a new .dmi file "unathi_extras.dmi" containing all the Unathi-specific hair styles and extras that were in use previously, and some additional extras that could be implemented later on if desired (Wiggling tails, tail spines and spikes, body markings, and snout shapes).

**Fixed**
- Unathi horns and frills, Tajaran hair styles and Skrell hair styles should no longer display a large ERROR on the preview, and should be visible ingame.
- Unathi, Tajaran and Skrell tails should now sync and display correct colour selected for body colour.
- Tails and wings from the Species Customization menu should now display properly when facing north.
- Tails and wings from the Species Customization menu should now hide accordingly when behind objects such as chairs and blankets.

**Known Issues**

- Unathi horns and spines should be removed from "extra.dm" IF NOT WANTED. "extra.dm" is the hair section and Unathi hair is usually limited to frills. They are there in case extra customization is wanted for players. Frills are considered hair and should stay there.
- Most frills and horns have colouring issues. The ones in the hair styles section (extra.dm) that have issues are too bright. When applying "blend = ICON_MULTIPLY" to the long unathi frills to test fixing this, they became too dark.
- The horns and spikes in the Ears section (sprite_accessories_vr.dm) are also too bright. They are using "color_blend_mode = ICON_ADD". However, when changed to "ICON_MULTIPLY" they become too dark as well.
- "Double Unathi Horns" doesn't work on the Hair Style section. It works in the Ears section but they are full white and are unable to change colour.

:cl:
add: New horns and frills for Unathi.
fix: Unathi, Tajaran and Skrell species specific hair styles no longer display a large ERROR on the preview, and are visible in-game.
fix: Unathi, Tajaran and Skrell tails now display the correct colour selected for body colour.
fix: Tails and wings from the Species Customization menu now display properly when facing north.
fix: Tails and wings from the Species Customization menu now hide accordingly when behind objects such as chairs and blankets.
fix: Held object sprites no longer get overlapped by tail and wing sprites.
/:cl:

### CODE REVIEW

**Comment by ShadzFen** [comment]
**If this gets pulled and merged, please start an issue thread noting the following issues I could not fix:**

**Known Issues**
- Unathi horns and spines should be removed from "extra.dm" IF NOT WANTED. "extra.dm" is the hair section and Unathi hair is usually limited to frills. They are there in case extra customization is wanted for players. Frills are considered hair and should stay there.
- Most frills and horns have colouring issues. The ones in the hair styles section (extra.dm) that have issues are too bright. When applying "blend = ICON_MULTIPLY" to the long unathi frills to test fixing this, they became too dark.
- The horns and spikes in the Ears section (sprite_accessories_vr.dm) are also too bright. They are using "color_blend_mode = ICON_ADD". However, when changed to "ICON_MULTIPLY" they become too dark as well.

**Comment by EvilJackCarver** [comment]
The Unathi base sprite colouration issue in #19 was fixed in #22 as it was using the wrong files; I left the issue ticket open for the Zorren half of that ticket. If it's lore-based (and please indicate how, since I'm not well-versed on lore) then I can close that ticket. 

Also, I think multiply should be the correct angle of attack here rather than additive; the greyscaling on Virgo bodyparts is light-near-white which would make additive colouring go beyond white.

**Comment by ShadzFen** [comment]
> The Unathi base sprite colouration issue in #19 was fixed in #22 as it was using the wrong files; I left the issue ticket open for the Zorren half of that ticket. If it's lore-based (and please indicate how, since I'm not well-versed on lore) then I can close that ticket.

Oh, well if the body colouration was already fixed then this only fixes the Unathi tail sprite colouration, which was causing them to look much brighter, like they were dunked in radiation.
image
I asked the people in charge of the lore and they confirmed that Highlander Zorren are purposely meant to have a darker colour palette. I imagine the sprite has a secondary shader layer or something to make it look darker than the colour selected, but I'm not sure so don't quote me on that.

> Also, I think multiply should be the correct angle of attack here rather than additive; the greyscaling on Virgo bodyparts is light-near-white which would make additive colouring go beyond white.

Well I can change it all to multiply then. Some of them become darker than others but I it will probably look better than the bright radiated mess they are now.

> One question, though, does this fix the layering issue mentioned in #26?

Nah. The normal base Unathi tail is visible when facing north, but any other tails added via the extras doesn't work. I could look into that if you want.

**Comment by ShadzFen** [comment]
Went through them one by one and set the blending settings to MULTIPLY on the parts that benefit the most from it. 

I checked comparisons with some pictures I had of my own character on the character creator menu from years back on AEIOU code and it seems this issue has actually always been a thing. The "Long Unathi Frills" have always been brighter than the colour selected. 
image
_Old AEIOU character creator with same frill colouring issue showing brighter than the colour selected._


It's easy to work around most of it, since 9/10 times you can get the colour you want to get by setting it a little darker or brighter. The biggest issue is the horns, which fail to turn a dark enough shade to go black. The most black they will go is dark-grey. This wouldn't be too much of an issue if it didn't also influence the rest of the colours, so it becomes impossible to get a specific dark shade of any colour since it will only go so dark before just turning grey. This is most easily visible with colours like dark red and dark green.
image
image
image


Unfortunately I have no idea how to fix this. However, it's a thousand times better than the horns being too dark to actually give them any colour to begin with (which is what happened when I applied the MULTIPLY blend to them).

**Comment by EvilJackCarver** [comment]
The layering issue is indeed one of our higher priority issues. I'll look through this when I get up, I'm literally about to go to bed but if you've tested this and it works spawned-in, I see no problem in merging it.

But let me get sleep before I give it the approval review.

**Comment by ShadzFen** [comment]
Sure! And yeah, I tested most of them in-game and they all worked as intended.

File: code/modules/sprite_accessories/extra.dm [hunk 1]
```diff
@@ -98,59 +98,121 @@
 
 	una_spines_long
 		name = "Long Unathi Spines"
+		icon = 'icons/mob/unathi_extras.dmi'
 		icon_state = "soghun_longspines"
-		species_allowed = list(SPECIES_UNATHI, SPECIES_XENOCHIMERA, SPECIES_PROTEAN) //Xenochimera get most hairstyles since they're abominations.
+		species_allowed = list(SPECIES_UNATHI, SPECIES_XENOCHIMERA, SPECIES_PROTEAN)
```
Comment (EvilJackCarver) [inline code/modules/sprite_accessories/extra.dm:103]: Please don't remove comments needlessly

File: code/modules/organs/external/organ_icon.dm [hunk 2]
```diff
@@ -108,7 +108,7 @@ var/global/list/limb_icon_cache = list()
 			if(hair_style && (species.get_bodytype() in hair_style.species_allowed))
 				var/icon/hair = new/icon(hair_style.icon, hair_style.icon_state)
 				if(hair_style.do_colouration)
-					hair.Blend(rgb(owner.r_hair, owner.g_hair, owner.b_hair), ICON_ADD)
+					hair.Blend(rgb(owner.r_hair, owner.g_hair, owner.b_hair), ICON_MULTIPLY)
```
Comment (EvilJackCarver) [inline code/modules/organs/external/organ_icon.dm:111]: `//eclipse edit`

File: code/modules/sprite_accessories/extra.dm [hunk 3]
```diff
@@ -98,59 +98,121 @@
 
 	una_spines_long
 		name = "Long Unathi Spines"
+		icon = 'icons/mob/unathi_extras.dmi'
```
Comment (EvilJackCarver) [inline code/modules/sprite_accessories/extra.dm:101]: Need a `//begin eclipse edit` somewhere around here

File: code/modules/sprite_accessories/extra.dm [hunk 4]
```diff
@@ -98,59 +98,121 @@
...
+		name = "Unathi Aquatic Frills"
+		icon = 'icons/mob/unathi_extras.dmi'
+		icon_state = "m_frills_aqua"
 		species_allowed = list(SPECIES_UNATHI, SPECIES_XENOCHIMERA, SPECIES_PROTEAN)
+		blend = ICON_MULTIPLY
 
 	taj_ears
 		name = "Tajaran Ears"
+		icon = 'icons/mob/unathi_extras.dmi'
 		icon_state = "ears_plain"
```
Comment (EvilJackCarver) [inline code/modules/sprite_accessories/extra.dm:216]: Need an `//end eclipse edit` somewhere near here

File: code/modules/sprite_accessories/sprite_accessories_vr.dm [hunk 5]
```diff
@@ -386,6 +386,101 @@
 	color_blend_mode = ICON_MULTIPLY
 	extra_overlay = "tesharilowinner"
 
+/datum/sprite_accessory/ears/soghun_longspines // Unathi extras START. Delete this comment after testing.
```
Comment (EvilJackCarver) [inline code/modules/sprite_accessories/sprite_accessories_vr.dm:389]: `//begin eclipse edit` (though this is a _vr which is probably not gonna get overwritten, but still)

File: code/modules/sprite_accessories/sprite_accessories_vr.dm [hunk 6]
```diff
@@ -386,6 +386,101 @@
...
+	do_colouration = 1
+	color_blend_mode = ICON_ADD
+
+/datum/sprite_accessory/ears/facial_dorsalfrill
+	name = "Unathi Dorsal Frill"
+	desc = ""
+	icon = "icons/mob/unathi_extras.dmi"
+	icon_state = "facial_dorsalfrill"
+	do_colouration = 1
+	color_blend_mode = ICON_ADD
```
Comment (EvilJackCarver) [inline code/modules/sprite_accessories/sprite_accessories_vr.dm:484]: `//end eclipse edit`

File: code/modules/mob/living/carbon/human/species/station/station.dm [hunk 7]
```diff
@@ -26,6 +26,7 @@
 	deform = 'icons/mob/human_races/r_def_lizard_vr.dmi'		//Eclipse edit.
 	tail = "sogtail"
 	tail_animation = 'icons/mob/species/unathi/tail_vr.dmi'		//Eclipse edit.
+	tail_blend = ICON_MULTIPLY
```
Comment (EvilJackCarver) [inline code/modules/mob/living/carbon/human/species/station/station.dm:29]: Please use `//eclipse edit` so we can tell if an upstream conflicts with changes we made

File: code/modules/sprite_accessories/extra.dm (hunk 1 above)
Comment (ShadzFen) [reply code/modules/sprite_accessories/extra.dm:103]: Oh shit I must have done that without realizing, my bad.

**Comment by EvilJackCarver** [comment]
Don't-merge: I think I've got something on the layering issue. About to apply and test.

File: code/modules/mob/living/carbon/human/update_icons.dm [hunk 8]
```diff
@@ -1258,7 +1258,7 @@ var/global/list/damage_icon_parts = list()
 
 	var/image/vr_wing_image = get_wing_image()
 	if(vr_wing_image)
-		vr_wing_image.layer = (-100)+WING_LAYER
```
Comment (EvilJackCarver) [inline code/modules/mob/living/carbon/human/update_icons.dm:1261]: Okay, here's the result of my research: Adding -100 to the layering on humans does make all this fit behind chairs and the like.

image
image

(everything was coloured to stand out easily)

However, this comes with two unfortunate side effects: 

1.) The tail layer is back behind the body, regardless of faced direction.
2.) Everything renders under the jumpsuit.

I've got dinner, and I'm down to a couple of tricks left up my sleeve. If they don't work, then I'm out of ideas.

**Comment by ShadzFen** [comment]
My previous comment's known issues have been solved courtesy of @EvilJackCarver. 

- Wings and tails form the Species Customization menu now hide properly under bed covers and chairs
image image


- Held items now properly display over those same wings and tail layers.
image image image

**Comment by drexample** [comment]
waiting on CL edit

**Comment by ShadzFen** [comment]
CL edited and ready! I think!

**Pull request merged**